### PR TITLE
Refactor tagging of TermFieldMatchData

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review
@havardpe : FYI

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/matching/handlerecorder.cpp [hunk 1]
```diff
@@ -112,6 +114,20 @@ HandleRecorder::add(TermFieldHandle handle,
     }
 }
 
+void
+HandleRecorder::tag_match_data(MatchData &match_data) {
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/matching/handlerecorder.cpp]: Nitpick: "{" should be on the next line.

**Review by geirst — APPROVED** [review]
👍

File: searchcore/src/vespa/searchcore/proton/matching/handlerecorder.cpp (hunk 1 above)
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/matching/handlerecorder.cpp]: For upcoming PR please add unit test for this in handle_recorder_test.cpp.

File: searchlib/src/vespa/searchlib/fef/termfieldmatchdata.cpp [hunk 2]
```diff
@@ -19,6 +20,7 @@ TermFieldMatchData::TermFieldMatchData() :
 TermFieldMatchData::TermFieldMatchData(const TermFieldMatchData & rhs) :
     _docId(rhs._docId),
     _fieldId(rhs._fieldId),
+    _flags(UNPACK_ALL_FEATURES_MASK),
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/fef/termfieldmatchdata.cpp]: should copy flags from rhs

**Pull request merged**